source/tools/CustomHostCpuManager.h

```diff
@@ -0,0 +1,477 @@
+     * played or the number of filters applied to a given signal.
+     * @param maxWorkUnits (not used) the maximum number of work units the
+     * application is planning to request.
+     */
+    void setApplicationLoad(int32_t currentWorkUnits,
+                            int32_t maxWorkUnits) override {
+        /*
+         * Perform an update only when the number of work units is changed, or
+         * if the mBWUpdateCounter is 0 (periodic update).
+         */
+        if (currentWorkUnits != getCurrentWorkUnits() ||
+                maxWorkUnits != getMaxWorkUnits() ||
+                mBWUpdateCounter == 0) {
+            int64_t expectedRuntime_ns;
+            double bandwidth;
+
+            /*
+             * mBWUpdateCounter is reset, to prevent a close bandwith updates
+             * due to the timer, after a workUnit change. This allows the
+             * statistics for the new workUnits to stabilize.
+             */
+            if (mBWUpdateCounter != 0)
+                mBWUpdateCounter = 0;
+
+            expectedRuntime_ns = mCpuTiming.getWorkUnitRuntime(currentWorkUnits);
+
+            /*
+             * When we read a 0 runtime for != 0 workUnits, it means that
+             * something bad is going on, so, set the maximum runtime.
+             * This may happen when the linear regression fails.
+             */
+            if (expectedRuntime_ns == 0 && currentWorkUnits != 0)
+                bandwidth = mMaxBW;
+            else
+                bandwidth = static_cast<double>(expectedRuntime_ns) / mPeriod_ns;
+
+            /* Add some offset to the computed bandwidth */
+            bandwidth *= 1.1;
+            bandwidth += 0.05;
+
+            /* Bound the bandwidth to the limit set by the Kernel */
+            if (bandwidth > mMaxBW)
+                bandwidth = mMaxBW;
+
+            if (currentWorkUnits != getCurrentWorkUnits() ||
+                    maxWorkUnits != getMaxWorkUnits()) {
+                /* Print the current status only when the workUnits changed */
+                printf("workUnits=%d\trequestDLBandwidth=%f",
+                       currentWorkUnits, bandwidth);
+                printf("\t[");
+                int width = 30;
+                int i;
+                for (i=0; i<bandwidth * width; ++i)
+                    printf("#");
+                for (; i<width; ++i)
+                    printf("_");
+                printf("]");
+                printf("\n");
+                fflush(stdout);
+            }
+
+            expectedRuntime_ns = mPeriod_ns * bandwidth;
+            updateDeadlineParams(expectedRuntime_ns,
+                                 mDeadline_ns,
+                                 mPeriod_ns);
+        }
+        HostCpuManagerBase::setApplicationLoad(currentWorkUnits, maxWorkUnits);
+    }
+
+    /**
+     * Update the current deadline parameters both of the scheduler and the
+     * class members.
+     */
+    int updateDeadlineParams(uint64_t runtime_ns,
```

Comment:
mixing Android and kernel naming conventions

Reply:
Not sure what would be the proper nomenclature...